haystack/components/rankers/sentence_transformers_diversity.py

```diff
@@ -119,16 +216,18 @@ def to_dict(self) -> Dict[str, Any]:
         return default_to_dict(
             self,
             model=self.model_name_or_path,
+            top_k=self.top_k,
             device=self.device.to_dict(),
             token=self.token.to_dict() if self.token else None,
-            top_k=self.top_k,
             similarity=self.similarity,
             query_prefix=self.query_prefix,
-            document_prefix=self.document_prefix,
             query_suffix=self.query_suffix,
+            document_prefix=self.document_prefix,
             document_suffix=self.document_suffix,
             meta_fields_to_embed=self.meta_fields_to_embed,
             embedding_separator=self.embedding_separator,
+            strategy=self.strategy,
```

Comment:
I think that both `strategy` and `similarity` should be converted into native Python types (`str`) to be YAML-serializable.

This fails now:
```python
from haystack.components.rankers import SentenceTransformersDiversityRanker
from haystack import Pipeline

pipe = Pipeline()
ranker = SentenceTransformersDiversityRanker(model="sentence-transformers/all-MiniLM-L6-v2", similarity="cosine")

pipe.add_component("ranker", ranker)

print(pipe.dumps())
```

The conversion to string can be done as follows:
- creating a `__str__` method in the `Enum` (e.g. in `Strategy`)
- using `str(self.strategy)` when calling `default_to_dict`.

You can find an example in the `DocumentJoiner`: https://github.com/deepset-ai/haystack/blob/2cc45dd5b9d8c0e9c326a0022f0c2f3775939313/haystack/components/joiners/document_joiner.py#L17